### DISCUSSION TITLE
SMS rate limiting requests/feedback

### DISCUSSION THREAD

**Discussion by anngbaum** [supabase/supabase#14264] [Feature Requests]
for context we have a native mobile app that uses SMS OTP for authentication exclusively. we saw a spike in signups and quickly hit the SMS rate limiting this morning (which I only discovered after finding [this PR](https://github.com/supabase/supabase/pull/11061)). this raised a few quick points of feedback:

1) default should be more than 30/hour. totally just our perspective, but that seemed shockingly low to us. I realize that the rate limit is probably in place for spam, but we have spend management well handled on the Twilio side so additional throttling feels a little unnecessary/restrictive, and the risk of declining good users seems very high.

2) logs for SMS OTP rate limiting show up as `info` statements instead of errors. not sure if this intended (in which case I just found it very unintuitive!) or a bug. this definitely slowed down our ability to identify the issue.

<img width="533" alt="Screen Shot 2023-05-10 at 8 50 16 AM" src="https://github.com/supabase/supabase/assets/29735378/aa2f2183-f9c5-4461-aac4-9b2644865607">

3) while it's great to see # of requests, for reporting purposes I really, really, really care about errors 1000x more than anything else. the problem here was both that these were not logging as proper errors, and most of the graphs/reporting just show volume of requests with no indication of errors or error rate. In a true wish list, I would be extremely delighted to receive some sort of email/trigger/anything if we exceed some number of errors, or hit our SMS rate limit, or just about anything that indicates that users who have downloaded our app can't get in it! as it stands, I don't think the reports page offers that, and so I find myself almost never using it.

<img width="948" alt="Screen Shot 2023-05-10 at 8 56 17 AM" src="https://github.com/supabase/supabase/assets/29735378/3119d3af-369d-4a2f-bed8-6f4518541b06">
<img width="569" alt="Screen Shot 2023-05-10 at 8 52 01 AM" src="https://github.com/supabase/supabase/assets/29735378/ead4664c-539a-44df-9711-d69b53812f3a">

**Comment by corepay** [comment]
![image](https://github.com/supabase/supabase/assets/48992923/3a36f382-686a-4571-a388-f2d12fd070bf)

Browsing through Supabase discussion (and issues) I see a lot of very important posts on bugs and features 'unanswered'.  Couple of them specific to my decision to use Supabase in a paid Production environment.  At this point leaning to self-host and hack my way through answers to these 'unanswered' posts or just use supabase as a 'low/no-code builder' TOOL and not a platform.

Wonder what they are doing with that $116M+ capital raised so far?

**Reply by anngbaum** [reply]
for what it's worth – when I initially filed this I simultaneously had a really helpful conversation with @saltcod on Twitter about the low rate limit, which he immediately circulated with the auth team. he shared their perspective on spam/reasoning for low defaults, as well as the "production checklist" [guide](https://supabase.com/docs/guides/platform/going-into-prod#auth-rate-limits) which calls this out. I believe that our app had launched well before the rate limiting was introduced, so we likely missed that update. I would assume that this is less of an issue for new projects. the logging/analytics feedback is still outstanding, and would be worth resurfacing at some point.

to provide one high-level data point – as a long-time customer of their platform product, I can definitely confirm that at I've been really happy with the responsiveness of the team across the board, and have actively encouraged a number of friends to use supabase for their own work!
